activestorage/app/javascript/activestorage/ujs.js

```diff
@@ -73,3 +74,14 @@ function disable(input) {
 function enable(input) {
   input.disabled = false
 }
+
+function tagClickedButton(event) {
+  event.target.setAttribute("data-clicked", "true")
+}
+
+document.addEventListener("turbolinks:load", function(event) {
```

Comment:
We can't rely Turbolinks events since not all apps use it. I'd add a new `click` listener here:
https://github.com/rails/rails/blob/ba1dab1e3b32a7c81cb9b8bdc22429f6620a3833/activestorage/app/javascript/activestorage/ujs.js#L10-L11

The handler can check if the `event.target` is a submit button and record the last clicked button in a `WeakMap` using the button's form as the key.